### DISCUSSION TITLE
Close HoverCard when a Menu.Item inside it is clicked

### DISCUSSION THREAD

**Discussion by LorenzoScebba** [mantinedev/mantine#3531] [Q&A] [answered]
Hi,

I'm trying to make it so that my HoverCard.Dropdown component closes automatically when a Menu.Item inside it is clicked.

At the moment this is the behaviour I'm getting OOTB: https://codesandbox.io/s/old-cherry-ezdxli?file=/src/App.js

Is there any simple way to do this since the HoverCard component cannot be controlled?

Thanks in advance!

**Comment by rtivital** [comment, marked as answer]
Use Popover component instead, it supports controlled opened state. This way you can have full control of component behavior.

**Reply by LorenzoScebba** [reply]
Thank you! For anyone wondering here's an example on how to do it: https://codesandbox.io/s/mantine-hover-card-using-popover-s9tgu9?file=/src/App.js

You can also implement a custom version of useDebouncedValue to force set the value so that the Popover closes immediatly instead of after X milliseconds.

Thank you @rtivital
